src/pages/sitemap/sitemap-index.xml.ts

```diff
@@ -0,0 +1,55 @@
+import fs from 'fs';
+import path from 'path';
+import { getServerSideSitemap } from 'next-sitemap';
+import { GetServerSideProps } from 'next';
+import { getAllContentItems } from '../../service/get-all-content';
+import { getTopicPagePathMappings } from '../../service/get-topic-paths';
+
+const DEVCENTER_URL =
+    process.env.DEVCENTER_URL || 'https://mongodb.com/developer';
+
+export const getServerSideProps: GetServerSideProps = async context => {
+    const curDate = new Date().toISOString();
+
+    const { topicContentTypePaths, topicPaths } =
+        await getTopicPagePathMappings();
+    const allContent = await getAllContentItems();
```

Comment:
How fresh does the sitemap need to be? `getAllContentItems` will make a heavy query, but we have that data pre-evaluated (example usage here: https://github.com/mongodb/devcenter/blob/f7d9407216e9d1f26ee477eff6b4894665d9ac1e/src/page-templates/topic-page/topic-page-data.ts#L10) so that in a way, it is static data "cached" in Next.js for future usage.

Reply:
wasn't aware of this, nice!